core/user_api.php

```diff
@@ -318,65 +326,42 @@ function user_ensure_email_unique( $p_email, $p_user_id = null ) {
 /**
  * Check if realname is specified, a unique real name, and doesn't match a username of another user.
  *
- * @param string $p_username The username to check.
  * @param string $p_realname The realname to check.
- * @return integer 0 realname matches another user's username, 1 no conflicts, >1 realname not unique.
+ * @param integer|null $p_user_id The user id that is allowed to conflict, otherwise, null.
+ * @return bool true: unique, false otherwise.
  */
-function user_is_realname_unique( $p_username, $p_realname ) {
+function user_is_realname_unique( $p_realname, $p_user_id = null ) {
 	if( is_blank( $p_realname ) ) {
 		# don't bother checking if realname is blank
-		return 1;
+		return true;
 	}
 
-	$p_username = trim( $p_username );
 	$p_realname = trim( $p_realname );
 
-	# allow realname to match username
-	$t_duplicate_count = 0;
-	if( $p_realname !== $p_username ) {
-		# check realname does not match an existing username
-		#  but allow it to match the current user
-		$t_target_user = user_get_id_by_name( $p_username );
-		$t_other_user = user_get_id_by_name( $p_realname );
-		if( ( false !== $t_other_user ) && ( $t_target_user !== $t_other_user ) ) {
-			return 0;
-		}
-
-		# check to see if the realname is unique
-		db_param_push();
-		$t_query = 'SELECT id FROM {user} WHERE realname=' . db_param();
-		$t_result = db_query( $t_query, array( $p_realname ) );
+	# check realname does not match an existing realname
+	$t_existing_user_id = user_get_id_by_name( $p_realname );
+	if( $t_existing_user_id !== false && ( $p_user_id === null || (int)$t_existing_user_id !== $p_user_id ) ) {
+		return false;
+	}
 
-		$t_users = array();
-		while( $t_row = db_fetch_array( $t_result ) ) {
-			$t_users[] = $t_row;
-		}
-		$t_duplicate_count = count( $t_users );
-
-		if( $t_duplicate_count > 0 ) {
-			# set flags for non-unique realnames
-			if( config_get( 'differentiate_duplicates' ) ) {
-				for( $i = 0;$i < $t_duplicate_count;$i++ ) {
-					$t_user_id = $t_users[$i]['id'];
-					user_set_field( $t_user_id, 'duplicate_realname', ON );
```

Comment:
As `differentiate_duplicates` defaulted to `OFF` and was not completely implemented, we could drop the column now. If not, we should create an issue with tag `schema` to consider it when introducing other database schema changes.

Reply:
I could not find a `differentiate_duplicates` in `schema.php`, so not sure if this code ever worked.

Reply:
You are right, the column does not exist.